### PR TITLE
refactoring pipelinerunstate - no logic changed at all

### PR DESCRIPTION
**Pull request by pritidesai**
# Changes

This is a "pure" refactoring i.e. no logic changes at all. 

- Converting exported methods into unexported local methods since they are being called from the same file/package to avoid direct access.
- Grouping exported and unexported (local) methods
- Adding comments to those methods

[Diff](https://github.com/tektoncd/pipeline/pull/3326/files?diff=split&w=1) with hiding spaces renders much better 🙏  

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```release-note
NONE
```

/kind cleanup

### CODE REVIEW

File: pkg/reconciler/pipelinerun/resources/pipelinerunstate.go [hunk 1]
```diff
@@ -57,10 +58,62 @@ func (state PipelineRunState) IsBeforeFirstTaskRun() bool {
...
+		status[rprt.TaskRunName] = prtrs
+	}
+	return status
+}
+
+// getNextTasks returns a list of tasks which should be executed next i.e.
 // a list of tasks from candidateTasks which aren't yet indicated in state to be running and
 // a list of cancelled/failed tasks from candidateTasks which haven't exhausted their retries
-func (state PipelineRunState) GetNextTasks(candidateTasks sets.String) []*ResolvedPipelineRunTask {
+func (state PipelineRunState) getNextTasks(candidateTasks sets.String) []*ResolvedPipelineRunTask {
```
Comment (jerop) [inline pkg/reconciler/pipelinerun/resources/pipelinerunstate.go:116]: should this also be moved to be grouped with the other unexported methods? 🤔

**Review by jerop — COMMENTED** [review]
Thanks for taking on the refactoring @pritidesai! 😄

**Review by sbwsg — APPROVED** [review]

File: pkg/reconciler/pipelinerun/resources/pipelinerunstate.go (hunk 1 above)
Comment (pritidesai) [reply pkg/reconciler/pipelinerun/resources/pipelinerunstate.go:116]: Hey @jerop `pipelinerunstate.go` holds attributes to `PipelineRunState` and `PipelineRunFacts`. Have grouped methods based on the receivers:

`PipelineRunState` exported methods
`PipelineRunState` unexported methods
`PipelineRunFacts` exported methods
`PipelineRunFacts` unexported methods

Comment (jerop) [reply pkg/reconciler/pipelinerun/resources/pipelinerunstate.go:116]: sounds great 👍

**Pull request merged**